### ISSUE TITLE
[ENH] Investigate where mocking can be used in unit tests to improve flakiness, speed up test suite

### ISSUE THREAD

**Issue by peytondmurray** [conda-incubator/conda-store#816]
### Context

Currently there are many _unit_ tests which are not very isolated. For example many tests reach out to conda-forge, and are therefore at risk of producing false failures for any number of network-related issues that are unrelated to the code being tested.

This issue is to plan and potentially track the work needed to mock out external dependencies, better isolating the unit tests to the code that we actually want to test.

One suggestion has been to run our own package repository locally, making it unnecessary for conda-store to make requests to conda-forge.

### Value and/or benefit

The test suite is currently flaky. This should help reduce flakiness and has the potential to greatly speed up testing, as waiting on external resources takes a lot of time.

**Comment by trallard** [comment]
Marking as in progress since @peytondmurray  is currently working on this

**Comment by peytondmurray** [comment]
We've seen significant (>60%) reduction in test times from targeting a few key areas. We can always do better, but the slowness in testing that led to the creation of this issue has largely been resolved. For any flaky tests in the future, let's open a new issue.
